main.js

```diff
@@ -0,0 +1,141 @@
+/* eslint no-param-reassign: ["error", { "props": false }] */
+const boxes = document.querySelectorAll('.box');
+
+const Player = (name, symbol) => ({ name, symbol });
+const player1 = Player('Player 1', 'X');
+const player2 = Player('Player 2', 'O');
+
+const gameBoard = (() => {
+  let board = new Array(9).fill(0);
+
+  const combos = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [2, 4, 6],
+    [0, 4, 8],
+  ];
+
+  const disable = () => {
+    boxes.forEach((box) => {
+      box.disabled = true;
+    });
+  };
+
+  const restart = () => {
+    boxes.forEach((box) => {
+      box.disabled = false;
+      box.value = '';
+    });
+    board = new Array(9).fill(0);
+  };
+
+  const turn = () => {
+    const zero = board.filter(a => a === 0);
+    if (zero.length % 2 === 0) {
+      return player2;
+    }
+    return player1;
+  };
+
+  const isFull = () => {
+    const zero2 = board.filter(a => a === 0);
+    if (zero2.length === 0) {
+      return true;
+    }
+    return false;
+  };
+
+  const checkWinner = () => {
```

Comment:
Related to my comment above on `turn` I was thinking you could simplify checking for a winner this way:

```javascript
// board factory
checkWinner  = (playerCombo) => {
  this.combos.some(combo => {
   combo.every(index => playerCombo.includes(index)
})
}
```
If all elements of one of the inner arrays of `combos` is in the `playerCombo` array (`playerCombo` is all the indices selected by a particular `player`) then this will return `true` -> meaning that player has won. 

Thus your player can now have an additional data:

```javascript
// player factory

const Player = (name, symbol) => {
  const moves = [];
  const turn = (index, board) => {
    board.addMark(index, this.symbol);
    this.moves.push(index);
  };
  return {name, symbol, moves, turn};
}
```